joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -224,23 +225,25 @@ controller_interface::return_type JointTrajectoryController::update(
         if (
           (before_last_point || first_sample) && *(rt_is_holding_.readFromRT()) == false &&
           !check_state_tolerance_per_joint(
-            state_error_, index, default_tolerances_.state_tolerance[index],
-            true /* show_errors */))
+            state_error_, index, active_tol->state_tolerance[index], true /* show_errors */) &&
+          *(rt_is_holding_.readFromRT()) == false)
```

Comment:
```suggestion
            state_error_, index, active_tol->state_tolerance[index], true /* show_errors */))
```

Do we need this `rt_is_holding_` check again after this?. I see it before the `check_state_tolerance_per_joint` call

Reply:
I'm not sure if I understand your comment, but yes we need the check here otherwise the tolerances will be checked with the holding position/final point after success.

Reply:
```
         if ( 
            (before_last_point || first_sample) && *(rt_is_holding_.readFromRT()) == false && 
            !check_state_tolerance_per_joint( 
             state_error_, index, default_tolerances_.state_tolerance[index], 
             true /* show_errors */)) 
             state_error_, index, active_tol->state_tolerance[index], true /* show_errors */) && 
           *(rt_is_holding_.readFromRT()) == false)
```

The `*(rt_is_holding_.readFromRT()) == false)` check is repeated. That's what I mean.

Reply:
ah you are right.. I had a look on the wrong branch yesterday :see_no_evil: must come from some merge of master